### PR TITLE
Clean up updated hypot function

### PR DESCRIPTION
**Pull request by musm**
@cfborges Looks like you forgot to delete the existing code   (ref https://github.com/JuliaLang/julia/pull/31922)

### CODE REVIEW

File: base/math.jl [hunk 1]
```diff
@@ -547,8 +547,8 @@ julia> hypot(3, 4im)
 ```
 """
 hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-hypot(x::Complex, y::Complex) = hypot(promote(abs(x),abs(y))...)
-hypot(x::Integer, y::Integer) = hypot(promote(float(x), float(y))...)
+hypot(x::Complex, y::Complex) = hypot(abs(x), abs(y))
+hypot(x::Integer, y::Integer) = hypot(float(x), float(y))
```
Comment (musm) [inline base/math.jl]: I think we should actually have  `hypot(x::T, y::T) where {T<:Number} = hypot(float(x), float(y)`   and removing the `Integer` method case.

Comment (oscardssmith) [reply base/math.jl]: Won't that break bigfloats et. al?

Comment (musm) [reply base/math.jl]: Right...

**Comment by cfborges** [comment]
> 
> 
> @cfborges Looks like you forgot to delete the existing code (ref #31922)

Actually, I purposely deferred that to someone else because of my previously demonstrated unfamiliarity with git (I commented on that somewhere in the original thread). I didn't want to make a mess out of it and waste everyone's time. But your point is excellent, the old code should come out.

**Review by cfborges — APPROVED** [review]

File: base/math.jl (hunk 1 above)
Comment (musm) [reply base/math.jl]: No it won't actually. `hypot` has specialized methods in MPFR.

**Comment by fredrikekre** [comment]
```
julia> hypot(3, 4im)
ERROR: StackOverflowError:
Stacktrace:
 [1] hypot(::Complex{Float64}, ::Complex{Float64}) at ./math.jl:550 (repeats 80000 times)
```

**Comment by musm** [comment]
@cfborges  Got it, thanks for the clarification. 
Should the method signature be restricted to `IEEEFloat`, instead of the more generic `AbstractFloat`?  Does the algorithm explicitly depend upon the representation of IEEE float? Glancing at the code I don't see anything that manipulates the bit field so assume it should be okay for all `AbstractFloats`, but it does depend on `eps` and I think the conditionals that depend on `eps` and `floatmin` may not hold for other 'generic' floats?

**Comment by cfborges** [comment]
> 
> 
> @cfborges Got it, thanks for the clarification.
> Should the method signature be restricted to `IEEEFloat`, instead of the more generic `AbstractFloat`? Does the algorithm explicitly depend upon the representation of IEEE float? Glancing at the code I don't see anything that manipulates the bit field so assume it should be okay for all `AbstractFloats`, but it does depend on `eps` and I think the conditionals that depend on `eps` and `floatmin` may not hold for other more 'generic' `AbstractFloats` types?

It should extend to any floating point system that uses round to nearest. The only things that really depend on the IEEE formats are the parts that prevent overflow/underflow. If someone has an abstract FPS that does something weird (like variable length significands) then it would have to be analyzed further to verify the accuracy. But at worst it would still be better than the previously existing hypot.

File: base/math.jl (hunk 1 above)
Comment (oscardssmith) [reply base/math.jl]: To me this still feels fishy as it means hypot will convert custom number types into floats if they don't specifically define a `hypot` function.

Comment (musm) [reply base/math.jl]: We already define `mathfunction(x::Real) = exp(float(x))` for numerous elementary math functions.

Comment (musm) [reply base/math.jl]: In any case we should probably change it from `T<:Number` to `T<:Real` .

**Review by stevengj — APPROVED** [review]

**Comment by musm** [comment]
good to merge?

**Pull request merged**